service/awsconfig/v10/resource/s3bucket/current.go

```diff
@@ -15,31 +17,45 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the S3 bucket")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the S3 buckets")
 
 	accountID, err := r.awsService.GetAccountID()
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	bucketName := key.BucketName(customObject, accountID)
-	headInput := &s3.HeadBucketInput{
-		Bucket: aws.String(bucketName),
-	}
-	_, err = r.clients.S3.HeadBucket(headInput)
-	if IsBucketNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the S3 bucket")
-		return BucketState{}, nil
-	}
-	if err != nil {
-		return BucketState{}, microerror.Mask(err)
+	bucketState := []BucketState{
+		BucketState{
+			Name:           key.TargetLogBucketName(customObject),
+			IsDeliveryLog:  true,
+			LoggingEnabled: true,
+		},
+		BucketState{
+			Name:           key.BucketName(customObject, accountID),
+			IsDeliveryLog:  false,
+			LoggingEnabled: true,
+		},
```

Comment:
This looks strange to me because the desired state is repeated in the current state. It's a pain that you can't list S3 buckets with a tag filter via the API.

I think it would be better to just have a slice of bucket names here and compare them against the desired state. WDYT?

Reply:
I think this is wrong in general. CurrentState shouldn't assume some values are set, those settings should be retrieved. I mean `IsDeliveryLog` and `LoggingEnabled`